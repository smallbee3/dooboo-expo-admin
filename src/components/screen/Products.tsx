import React from 'react';
import styled from 'styled-components/native';

const Container = styled.View`
  background-color: ${({ theme }): string => theme.background};
  flex: 1;
  align-items: center;
  justify-content: center;
`;

const Title = styled.Text`
  font-weight: bold;
  color: ${({ theme }): string => theme.font};
`;

const Products: React.FC = () => {
  return (
    <Container>
      <Title>Products Screen</Title>
    </Container>
  );
};

export default Products;
