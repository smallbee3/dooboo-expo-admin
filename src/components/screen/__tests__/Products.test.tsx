import 'react-native';

import React, { ReactElement } from 'react';
import {
  RenderResult,
  render,
} from '@testing-library/react-native';
import {
  createTestElement,
  createTestProps,
} from '../../../../test/testUtils';

import Products from '../Products';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;
let testingLib: RenderResult;

describe('[Products] Rendering', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Products {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const { baseElement } = testingLib;
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('[Products] Interaction', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Products {...props} />);
    testingLib = render(component);
  });
});
