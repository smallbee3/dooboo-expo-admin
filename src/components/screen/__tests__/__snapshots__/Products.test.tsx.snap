// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Products] Rendering renders without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1B1B21",
              "fontWeight": "bold",
            },
          ]
        }
      >
        Products Screen
      </Text>
    </View>
  </View>
</View>
`;
